embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/properties/EmbraceSessionPropertiesService.kt

```diff
@@ -1,15 +1,20 @@
 package io.embrace.android.embracesdk.session.properties
 
+import io.embrace.android.embracesdk.capture.session.SessionPropertiesDataSource
 import io.embrace.android.embracesdk.ndk.NdkService
 
 internal class EmbraceSessionPropertiesService(
     private val ndkService: NdkService,
-    private val sessionProperties: EmbraceSessionProperties
+    private val sessionProperties: EmbraceSessionProperties,
+    private val sessionPropertiesDataSource: SessionPropertiesDataSource?
 ) : SessionPropertiesService {
 
     override fun addProperty(key: String, value: String, permanent: Boolean): Boolean {
         val added = sessionProperties.add(key, value, permanent)
         if (added) {
+            sessionPropertiesDataSource?.apply {
```

Comment:
There's a bug here where by any truncated key or value is not reflected in the session span because the validation is done at the EmbraceSessionProperties level. I will fix this tomorrow....